File: src/Modules/General/Ranks.cs
using Discord.Commands;
using System.Threading.Tasks;
using System.Linq;
using DEA.Common.Extensions;
namespace DEA.Modules.General
{
    public partial class General
    {
        [Command("Ranks")]
        [Summary("View all ranks.")]
        public async Task Ranks()
        {
            if (Context.DbGuild.RankRoles.ElementCount == 0)
            {
                ReplyError("There are no ranks yet!");
            }

            var description = string.Empty;
            foreach (var rank in Context.DbGuild.RankRoles.OrderBy(x => x.Value.AsDouble))
            {
                var role = Context.Guild.GetRole(ulong.Parse(rank.Name));
                if (role == null)
                {
                    await _guildRepo.ModifyAsync(Context.DbGuild, x => x.RankRoles.Remove(rank.Name));
                    continue;
                }
                description += $"{((decimal)rank.Value.AsDouble).USD()}: {role.Mention}\n";
            }

            await SendAsync(description, "Ranks");
        }
    }
}

File: src/commands/general/Ranks.js
const patron = require('patron.js');
const util = require('../../utility');

class Ranks extends patron.Command {
  constructor() {
    super({
      name: 'ranks',
      group: 'general',
      description: 'View all ranks in this guild.'
    });
  }

  async run(msg, args) {
    if (msg.dbGuild.roles.rank.length === 0) {
      return util.Messenger.replyError(msg.channel, msg.author, 'There are no rank roles yet!');
    }

    const sortedRanks = msg.dbGuild.roles.rank.sort((a, b) => a.cashRequired - b.cashRequired);

    let description = '';
    for (let i = 0; i < sortedRanks.length; i++) {
      const rank = msg.guild.roles.find((x) => x.id === sortedRanks[i].id);
      
      description+= rank + ': ' + util.NumberUtil.USD(sortedRanks[i].cashRequired) + '\n';
    }

    return util.Messenger.send(msg.channel, description, 'Ranks');
  }
}

module.exports = new Ranks();
